lib/spack/spack/test/stage.py

```diff
@@ -902,8 +917,91 @@ def test_diystage_preserve_file(self, tmpdir):
         with open(readmefn) as _file:
             _file.read() == _readme_contents
 
+    def test_create_cmakebuildstage(self, cmake_ext_build, mock_stage_archive):
+        """Test that cmake external build dir is created during stage creation"""
+        archive = mock_stage_archive()
+        with CMakeBuildStage("jk6dj", archive.url, name=self.stage_name) as stage:
+            # external build dir should already have been created at this point
+            # ensure symlink to build dir exists
+            assert str(stage.root_stage_context.name) == "spack-build-jk6dj"
+            assert "spack-build-jk6dj" in os.listdir(stage.path)
+            # check symlink points to the right place
+            assert str(stage.root_stage_context.resolve()) == str(cmake_ext_build / "a")
```

Comment:
I'm inclined to say that this should be testing `.build_directory` directly (or at least, we should be checking that in addition).